app/components/autocomplete/at_mention/index.ts

```diff
@@ -26,21 +28,48 @@ const enhanced = withObservables([], ({database, channelId}: WithDatabaseArgs &
 
     let useChannelMentions: Observable<boolean>;
     let useGroupMentions: Observable<boolean>;
+    let isChannelConstrained: Observable<boolean>;
+    let isTeamConstrained: Observable<boolean>;
+    let team: Observable<TeamModel | undefined>;
+
 
     if (channelId) {
         const currentChannel = observeChannel(database, channelId);
+        team = currentChannel.pipe(switchMap((c) => {
+            if (c && c.teamId) {
+                return c.team.observe() as Observable<TeamModel>;
```

Comment:
Why do we need the `as`?

Reply:
because model.anotherModel.observe() returns `any` as its type

Reply:
It shouldn't be. The type of `c.team` is a `Relation<TeamModel>`, and a `Relation<T>`, the `observe()` function is of type `Observable<T|null>`. So... we shouldn't need or be doing the casting.

Reply:
`model.observe()` returns `Model | null`, but our `observeModel()` methods return `Model | undefined` - and I think I was getting type errors because of the difference between `null` and `undefined` further down in the code.

I have cast the model type there as if the `channel.teamId` is available, there should be a related team we can observe. 

I'm not sure, but a better way might be to pipe/map the observable and see if the team actually exists, and return the correct `null` or `undefined` behaviour as expected further down? 0/5 on which way might be better.